### DISCUSSION TITLE
Capping frame rate?

### DISCUSSION THREAD

**Discussion by aleph1** [Kode/Kore#639] [Q&A]
I took a look through the source, but didn't notice a way to cap frame rate, is there one?

**Comment by aleph1** [comment]
It looks like there might be a few ways to implement this, and I am wondering if you have a preferred approach, or even any interest in this being added to the core. Display frame rate can be set on most systems (see below) but this doesn't bind the game's FPS to the system setting. I suppose we would need a sleep or spinlock system to wake at the right time to effectively draw the frame. Perhaps this is already implemented but after digging through the source even more I still couldn't find anything.

On Mac and iOS there is this: https://developer.apple.com/documentation/metalkit/mtkview/1536027-preferredframespersecond
On Android, this: https://developer.android.com/guide/topics/media/frame-rate

And I imagine there is something similar for Windows.

If you have any opinions on this please let me know, as it is something I would be interested in contributing as I am currently working on a prototype where I need to lock the game to 30fps for mobile.

**Reply by RobDangerous** [reply]
macOS, iOS and super recent versions of Android is not most systems. Sleeping and spinlocking is already possible so I see no need to add anything to Kinc here.
